packages/x-tree-view/src/internals/useTreeView/useTreeView.ts

```diff
@@ -18,6 +19,19 @@ import { useTreeViewModels } from './useTreeViewModels';
 import { TreeViewContextValue } from '../TreeViewProvider';
 import { TREE_VIEW_CORE_PLUGINS } from '../corePlugins';
 
+export function useTreeViewApiInitialization<T>(
+  inputApiRef: React.MutableRefObject<T> | undefined,
+): React.MutableRefObject<T> {
+  const publicApiRef = React.useRef({}) as React.MutableRefObject<T>;
+  if (inputApiRef && !inputApiRef.current) {
```

Comment:
This is not compatible with React concurrent mode, we should never update a ref in render except for initialization.

Is there a downside to just do the following?

```tsx
  const fallbackPublicApiRef = React.useRef({}) as React.MutableRefObject<T>;

  if (inputApiRef) {
    return inputApiRef
  }

  return fallbackApiRef
```